src/shared_modules/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -30,21 +33,41 @@ SQLiteDBEngine::~SQLiteDBEngine()
 }
 
 void SQLiteDBEngine::setMaxRows(const std::string& table,
-                                const unsigned long long maxRows)
+                                const int64_t maxRows)
 {
     if (0 != loadTableData(table))
     {
-        const constexpr auto ROW_COUNT_POSTFIX{"_row_count"};
-        const std::string sql
+        std::lock_guard<std::mutex> lock(m_maxRowsMutex);
+
+        if (maxRows < 0)
+        {
+            throw dbengine_error { MIN_ROW_LIMIT_BELOW_ZERO };
+        }
+        else if (0 == maxRows)
+        {
+            m_maxRows.erase(table);
+        }
+        else
         {
-            maxRows ?
-            "CREATE TRIGGER " + table + ROW_COUNT_POSTFIX + " BEFORE INSERT ON " + table +
-            " WHEN (SELECT COUNT(*) FROM " + table + ") >= " + std::to_string(maxRows) +
-            " BEGIN SELECT RAISE(FAIL, '" + SQLite::MAX_ROWS_ERROR_STRING + "'); END;"
-            : "DROP TRIGGER " + table + ROW_COUNT_POSTFIX
+            const auto stmt
+            {
+                getStatement("SELECT COUNT(*) FROM " + table + ";")
+            };
 
-        };
-        m_sqliteConnection->execute(sql);
+            if (stmt->step() == SQLITE_ROW)
+            {
+                const auto currentRows
+                {
+                    stmt->column(0)->value(int64_t{})
+                };
+
+                m_maxRows[table] = { maxRows, currentRows };
```

Comment:
How about the case that we are shrinking the database? i.e current database hold 100 rows and we set its max to 90, leading to 10 out of bound rows and no warning? this also will create infinite scenario even when we need to eliminate rows **https://github.com/wazuh/wazuh/pull/13192/files#diff-e02c6074db104df300a9237ef9da28ecef6fff8c5fc1359c34cc9f39965e8e26R2076-R2079**

Reply:
the idea is that it be a limit that does not allow insertions above the limit, not that it shrinks, since this should generate rows deletion events, to be sent to the manager for deltas. This would be an overwork for the manager, without need.
When we make the hot configuration changes we can review it.